### DISCUSSION TITLE
A lot of standard copy and paste code not working in templater for me.

### DISCUSSION THREAD

**Discussion by chump** [SilentVoid13/Templater#969] [Help]
I'm not sure what's going on, but simple snippets of code that many people are using aren't translating into read mode for me.

# <% tp.file.title %>

# <% moment(tp.file.title,'YYYY-MM-DD').format("dddd, MMMM DD, YYYY") %>

for example. It simply shows up as plain text rather than running the code.

**Comment by chump** [comment]
I figured out the second line with the moment issue - I had to add a + after the % for some odd reason. 

However basic code like this still isn't working
`# <% tp.file.title %>`
`alias: <% tp.file.title %>`
`<% tp.file.cursor(1) %>`

It seems to just load up as straight text for me.

**Reply by AB1908** [reply]
You need to replace this using the Replace templates in active file command or whatever it's called. Can you tell us if this is unclear from the documentation so that we can tweak it?

**Reply by OogieM** [reply]
Not the OP But I have the same problem and I looked in Documentation and can't find where how to make it update is discussed at all. So Yeah, something needs to be clearer. I'd like to see it in the FAQ but since I still can't make it work I can’t even offer any suggestions on how to write the FAQ question/answer.

**Reply by AB1908** [reply]
The template doesn't resolve itself if you copy and paste. You need to either insert that text as a template or replace templates in the active file. Take a look at the Obsidian community talk that happened in the recent Obsidian October for examples.

**Reply by OogieM** [reply]
So stil not quite getting it. Here is my procedure. create a new note, command t to insert a template, select one from my set of templates. It comes in with the code but is not executed.  So the variables are not filled. 

Searched for replace in the documentation and didn't find anything that looked like it explained what to do.

And another stupid user question Obsidian talk in Obsidian October? Link or reference please? No clue what or where that is.

**Reply by chump** [reply]
I appreciate the responses here. I tentatively seem to have things working, although I don't know what changed. I've been fiddling with this for days. And tried uninstalling, resetting multiple times. Some of the tp. commands would render and others wouldn't. I wonder if there's a certain order to installation, or if it was the latest Obsidian update, I'm glad I've got it working, but really confused by the inconsistent experience earlier.

**Reply by AB1908** [reply]
If it's still unclear after watching a video or two, I can spare some time to walk through it on a call.

**Reply by AB1908** [reply]
https://youtu.be/-PYyO7y0aBs

**Comment by chump** [comment]
One thing that maybe I'm missing is that within for example my default note template - I have 

alias: <% tp.file.title %>
# <% tp.file.title %>

it runs correctly the first time (triggers on new note creation), but the new note by default is 'untitled' so it takes on that same title. After I've updated the document name it doesn't seem to automatically update these parts? I've tried running templater again (replace templates in the active file) and nothing seems to change it from 'untitled'. 

Is there a concept or a checkbox I'm missing here?

**Reply by AB1908** [reply]
Yes, it only works once on insertion and does not dynamically update. Were you looking for dataview instead?
